### PR TITLE
Add an overengineered AWS example

### PR DESCRIPTION
**Pull request by vkleen**

### CODE REVIEW

File: examples/aws/modules/aws-simple-ec2.ncl [hunk 1]
```diff
@@ -0,0 +1,71 @@
+let Aws = import "./types.ncl" in
+let Utils = import "../utils.ncl" in
+{
+  aws.simple.ec2.instances
+    | {
+      _ : {
```
Comment (yannham) [inline examples/aws/modules/aws-simple-ec2.ncl:6]: Should you use a contract here? (remember that now dictionary type kill e.g. recursive records)

```suggestion
      _ | {
```

File: examples/aws/utils.ncl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  merge_array | Array { _ : Dyn } -> { _ : Dyn }
+    = fun rs => std.array.fold_left (fun r x => r & x) {} rs,
```
Comment (yannham) [inline examples/aws/utils.ncl:3]: I think you can use `std.array.merge_all` now?

File: examples/aws/utils.ncl [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+  traceVal = fun x => std.trace (std.serialize 'Json x) x,
```
Comment (yannham) [inline examples/aws/utils.ncl:5]:
```suggestion
  trace_value = fun x => std.trace (std.serialize 'Json x) x,
```

**Review by yannham — APPROVED** [review]
Mostly LGTM, but I think it could do with a README or in-code documentation at least, because there's several fairly non trivial elements (why do you need to use bare Terraform interpolation, why do you need to use those `${null}`, etc.)

**Comment by yannham** [comment]
@vkleen I had to check something on the repo, and notice this stale PR. What should we do with it?

**Comment by vkleen** [comment]
Thanks for reminding me about that! I guess it's valuable as an experiment, if it still works. I'll test it (updating e.g. the dictionary types into contracts) and if it works fine, I'll merge it. Otherwise it's probably best to just close it.

**Pull request merged**